File: lectures/code/cc_04_00a.py
import math

class Point:
    __x = 0.0
    __y = 0.0

    def __init__(self, x, y):
        self.__x = x
        self.__y = y

    def dump(self):
        print('Object point@%x x=%f y=%f' % (id(self),self.__x,self.__y))

pt = Point(4.0, 5.0)
Point.dump(pt)
del(pt)

File: lectures/code/cc_04_00a.cpp
#include <iostream>
#include <math.h>

class Point {
  private: 
    double x, y;

  public: 
    Point(double xc, double yc)  {
        x = xc;
        y = yc;
    };

    void dump() {
        printf("Object point x=%f y=%f\n", x, y);
    }
};

int main() {
    Point pt(4.0, 5.0);
    pt.dump();
}

// rm -f a.out; g++ cc_04_00a.cpp; a.out; rm -f a.out

